soc/arm/st_stm32/stm32l1/Kconfig.defconfig.stm32l151xc

```
@@ -0,0 +1,14 @@
+# ST Microelectronics STM32L151XC MCU
+
+# Copyright (c) 2020 Abel Sensors
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32L151XC
+
+config SOC
+	default "stm32l151xc"
+
+config NUM_IRQS
+	default 53
```

```suggestion
	default 57
```
For simplicity, we apply following formula: highest IRQ number +1

---

What is the number 57 (56+1) based on? The datasheet states the NVIC can handle up to 53 maskable interrupt channels (https://www.st.com/resource/en/datasheet/stm32l151cc.pdf page 19), so that's the information that I based the number on.

---

This is based on the number provided in CMSIS file:
https://github.com/zephyrproject-rtos/hal_stm32/blob/cea57f86d3ade46c0b99dab32b2ac63435ed0693/stm32cube/stm32l1xx/soc/stm32l151xc.h#L134

---

Okay, yes makes sense to base it on the amount in the HAL, I'll change it.
I'll add a comment to ensure that it is clear what it is based on, since there is a difference between the datasheet and HAL.